### PR TITLE
Rust 2024 call for testing

### PR DESCRIPTION
**Pull request by ehuss**
This adds a post to ask people to help with the 2024 edition testing.

Post-dated for Monday. I hope to at least have cargo merged tomorrow. There's also a bunch of doc updates, but those probably won't be ready before Monday, so not sure if I should wait another day.

cc @traviscross 

[Rendered](https://github.com/ehuss/blog.rust-lang.org/blob/2024-call-for-testing/posts/2024-11-27-Rust-2024-public-testing.md)

### CODE REVIEW

**Comment by weihanglo** [comment]
@ehuss which PR needs to be included in Cargo for testing 2024 edition?

**Comment by ehuss** [comment]
https://github.com/rust-lang/cargo/pull/14828 (and there are some updates I need to push to that as well), along with updating the submodule in rust-lang/rust (which I can do tonight), and then waiting for that to hit nightly.

**Comment by traviscross** [comment]
Status: I've revised the post a bit.  It looks like we're still waiting on https://github.com/rust-lang/cargo/pull/14828 to merge and end up in the nightly toolchain.  It does make a much cleaner announcement and a better experience for people testing if we let that merge, so we'll see if that can happen.  Perhaps this can go out tomorrow.  That'd still be OK.

**Review by traviscross — APPROVED** [review]

File: posts/2024-11-27-Rust-2024-public-testing.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+title: "Rust 2024 call for testing"
+author: "Eric Huss & TC"
+team: the Edition 2024 Project Group <https://doc.rust-lang.org/nightly/edition-guide/rust-2024/index.html>
+---
+
+# Rust 2024 call for testing
+
+We've been hard at work on Rust 2024. We're thrilled about how it has turned out. It's going to be largest edition since Rust 2015. It has a great many improvements that make the language more consistent and ergonomic, that further upon our relentless commitment to safety, and that will open the door to long-awaited features such as `gen` blocks, `let` chains, and the never (`!`) type. For more on the changes, see the nightly [Edition Guide](https://doc.rust-lang.org/nightly/edition-guide/rust-2024/index.html).
+
+As planned, we recently merged the feature-complete Rust 2024 edition [to the release train](https://github.com/rust-lang/rust/pull/133349) for Rust 1.85. It has now entered **nightly beta**.
```
Comment (weihanglo) [inline posts/2024-11-27-Rust-2024-public-testing.md:12]: Maybe it's just me, but this is my first time seeing the term "nightly beta". I realized it is already used in the edition 2024 schedule in edition book, though still a bit confused and can't get what that means.

**Comment by traviscross** [comment]
The new nightly is out; I'm testing and preparing to send this out.

File: posts/2024-11-27-Rust-2024-public-testing.md (hunk 1 above)
Comment (compiler-errors) [reply posts/2024-11-27-Rust-2024-public-testing.md:12]: presumably it means beta in the sense of"pre-release testing", not the beta branch of rust. I do think the term is a bit strange, and probably could be replaced, though.

Comment (traviscross) [reply posts/2024-11-27-Rust-2024-public-testing.md]: Thanks.  I added a footnote about this.

**Comment by traviscross** [comment]
OK.  It's now live.  Tells your friends and feel free to post to social media.

https://blog.rust-lang.org/2024/11/27/Rust-2024-public-testing.html

**Pull request merged**